examples/result_file_dataclasses.py

```diff
@@ -0,0 +1,202 @@
+"""
+Working with result files using Python-native dataclasses.
+"""
+from collections.abc import Iterator
+from operator import attrgetter
+from pathlib import Path
+
+import pendulum
+import polars as pl
+
+from indico import IndicoClient
+from indico.queries import (
+    GetSubmission,
+    SubmissionResult,
+    RetrieveStorageObject,
+    SubmitReview,
+)
+from indico_toolkit import results
+
+
+"""
+Load a result file.
+"""
+client = IndicoClient()
+submission = client.call(GetSubmission(123))  # Or similar.
+result_dict = client.call(RetrieveStorageObject(submission.result_file))
+result = results.load(result_dict)
+
+# If your submission is unreviewed,
+# generate a result file to include review information.
+submission_result = client.call(SubmissionResult(submission, wait=True))
+result_dict = client.call(RetrieveStorageObject(submission_result.result))
+result = results.load(result_dict)
+
+# Or convert the unreviewed submission to a reviewed submission.
+# Due to the lack of review data, only `document.final` is populated.
+result = results.load(result_dict, convert_unreviewed=True)
+
+# `results.load` also takes JSON strings and file paths.
+for result_file in Path("results_folder").glob("*.json"):
+    result = results.load(result_file)
+
+
+"""
+Save document classification and extractions to CSV.
+"""
+
+
+def get_predictions(result: results.Result) -> Iterator[dict[str, object]]:
+    yield {
+        "submission_id": result.submission_id,
+        "field": "Classification",
+        "value": result.document.final.classification.label,
+        "confidence": result.document.final.classification.confidence,
+    }
+
+    for extraction in result.document.final.extractions:
+        yield {
+            "submission_id": result.submission_id,
+            "field": extraction.label,
+            "value": extraction.text,
+            "confidence": extraction.confidence,
+        }
+
+
+pl.DataFrame(get_predictions(result)).write_csv("predictions.csv")
+
+
+"""
+Auto review predictions.
+"""
+pre_review = result.document.pre_review
+extractions = pre_review.extractions
+
+# Downselect based on highest confidence.
+for model, extractions in extractions.groupby(attrgetter("model")).items():
```

Comment:
for the examples, I think we're getting to customized in the behavior here to be generally useful (unless someone is reproducing this exact code)

Reply:
or perhaps consider incorporating `attrgetter` into the class itself

Reply:
I've split the example script out into three examples covering results to CSV, a minimal auto review, and loading and traversing results. Hopefully that makes them more focused and useful as real executable examples.

Highest confidence downselection was just one of the common use cases for auto review that also showed off quite a bit of the results API. Definitely open to ideas for the auto review example that are less customized!